### DISCUSSION TITLE
Vulkan Working and Operational

### DISCUSSION THREAD

**Discussion by devshgraphicsprogramming** [Devsh-Graphics-Programming/Nabla#473] [Q&A]
Things left to hammer out

#### `nbl::system::IAsyncQueueDispatcher` design failure and deadlocks

Being addressed in https://github.com/Devsh-Graphics-Programming/Nabla/compare/async_queues?expand=1

#### Destructors of certain objects are not threadsafe

Issue #472 

#### Built-in Resources Compilation failure and rework

TODO by @AnastaZIuk 

#### DXC doesn't compile on Linux

Need access to HLSL on Linux

#### Compilation under GCC and Clang on Linux

To be addressed by: https://github.com/Devsh-Graphics-Programming/Nabla/pulls/pollend

#### X11/XCB Windowing and UI on Linux

Basically non-headless Vulkan mode for Linux #462 

#### Swapchain Manager

The way we deal with resizes is very disorganized and could improve.

#### Less verbose CommonAPI

Basically a refactor

<hr /><em>This discussion was created from the release <a href='https://github.com/Devsh-Graphics-Programming/Nabla/releases/tag/v0.4-alpha'>Vulkan Working and Operational</a>.</em>
